options/posix/generic/pthread-stubs.cpp

```diff
@@ -214,9 +219,19 @@ int pthread_join(pthread_t thread, void **ret) {
 	return 0;
 }
 
-int pthread_detach(pthread_t) {
-	__ensure(!"Not implemented");
-	__builtin_unreachable();
+int pthread_detach(pthread_t thread) {
+	auto tcb = reinterpret_cast<Tcb*>(thread);
+	if (!(tcb->selfPointer == tcb))
+		return ESRCH;
+	if (!tcb->isJoinable)
+		return EINVAL;
```

Comment:
This needs to do an `__atomic_load_n` (same for all other accesses to `isJoinable`).